src/main/jbake/content/blog/2017/02/13/monitoring-canary-deployments.adoc

```diff
@@ -0,0 +1,249 @@
+  "itemId" : "laptop",
+  "quantity" : 3,
+  "id" : "d3f09a97-6f87-45c5-98c8-b592b6beb501"
+}
+
+Creating an order for account=sarah item=car quantity=4
+Not account found
+
+Creating an order for account=steve item=car quantity=2
+{
+  "accountId" : "steve",
+  "itemId" : "car",
+  "quantity" : 2,
+  "id" : "7f095b1e-94a2-4398-8480-01aaea43e021"
+}
+...
+...
+```
+
+Hawkular APM should be located under the service name `hawkular-apm`, so, the URL
+be similar to this: `https://hawkular-apm-order-manager.192.168.2.107.xip.io`. Note
+that the final part of the hostname might be different.
+
+After a few minutes with our `genorders.sh` script running, we should be able to
+see a screen similar to this:
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-02-13-initial-apm-screen-with-data.png[Hawkular APM Dashboard with Data]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-02-13-initial-apm-screen-with-data.png[Hawkular APM Dashboard with Data]
+endif::[]
+
+Let's open the _Services_ tab, select the service _order-manager.account-manager_
+and the _order-manager.account-manager-1_ version. These names are derived automatically
+from the service name and deployment number provided by OpenShift. Let's select
+an _Aggregation Interval_ of 10 seconds.
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-02-13-single-service.png[Hawkular APM Services Tab with One Service]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-02-13-single-service.png[Hawkular APM Services Tab with One Service]
+endif::[]
+
+Now, we'll create a new build and new application for our "Blue" deployment. We
+first set the current deployment to have 4 replicas, so that we'll have 80% of
+the traffic going to the old version, and 20% going to the new one.
+
+```bash
+cd account-manager/
+oc scale deploymentconfig account-manager --replicas=4
+oc new-build --binary --name=account-manager-blue -l app=account-manager-blue
+oc start-build account-manager-blue --from-dir=. --follow
+oc new-app \
+  account-manager-blue \
+  -l app=account-manager-blue \
+  HAWKULAR_APM_URI='http://hawkular-apm' \
+  HAWKULAR_APM_USERNAME='admin' \
+  HAWKULAR_APM_PASSWORD='password'
+```
+
+An output similar to this is expected:
+```
+--> Found image 3f61c6a (37 seconds old) in image stream account-manager-blue under tag "latest" for "account-manager-blue"
+
+    * This image will be deployed in deployment config "account-manager-blue"
+    * The image does not expose any ports - if you want to load balance or send traffic to this component
+      you will need to create a service with 'expose dc/account-manager-blue --port=[port]' later
+
+--> Creating resources with label app=account-manager-blue ...
+    deploymentconfig "account-manager-blue" created
+--> Success
+    Run 'oc status' to view your app.
+```
+
+We should also have a new service displayed on OpenShift console. Note the new
+_account-manager-blue_ deployment, with 1 pod.
```

Comment:
You could put an orange circle around the -blue one on the image to make it easier to spot it.